src/Polly.Shared/Wrap/PolicyWrapSyntax.cs

```diff
@@ -84,37 +84,62 @@ public partial class Policy
         /// <param name="policies">The policies to place in the wrap, outermost (at left) to innermost (at right).</param>
         /// <returns>The PolicyWrap.</returns>
         /// <exception cref="System.ArgumentException">The enumerable of policies to form the wrap must contain at least two policies.</exception>
-        public static PolicyWrap Wrap(params Policy[] policies)
+        public static PolicyWrap Wrap(params ISyncPolicy[] policies)
         {
             switch (policies.Count())
             {
                 case 0:
                 case 1:
                     throw new ArgumentException("The enumerable of policies to form the wrap must contain at least two policies.", nameof(policies));
                 default:
-                    IEnumerable<Policy> remainder = policies.Skip(1);
-                    return policies.First().Wrap(remainder.Count() == 1 ? remainder.Single() : Wrap(remainder.ToArray()));
+                    ISyncPolicy[] remainder = policies.Skip(1).ToArray();
```

Comment:
Why add the overhead of creating an array using ToArray()? Use `IEnumerable<ISyncPolicy>`

Reply:
In the old code, the enumerable already was converted into an array when passing it as argument to the policies.First().Wrap(...) method. All I did was move the ToArray to an earlier moment, to prevent both the Count() and the ToArray() to navigate through the enumerable. It should result to less overhead rather than more.

If we want to remove the ToArray, I reckon these type of performance improvements should be part of a separate pull request?